core/src/worker/workflow/mod.rs

```diff
@@ -1273,3 +1304,22 @@ impl LocalActivityRequestSink for LAReqSink {
         res
     }
 }
+
+/// Sorts jobs in an activation to be in the order lang expects:
+/// `patches -> signals -> other -> queries`
+fn sort_act_jobs(wfa: &mut WorkflowActivation) {
+    wfa.jobs.sort_by(|j1, j2| {
```

Comment:
Can you confirm this is a "stable" sort (meaning equal things aren't reordered)?

Reply:
Yes, it's stable.